src/main/java/org/jenkinsci/plugins/mesos/NonConfigurableMesosCloud.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015 CloudBees, Inc. and other contributors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jenkinsci.plugins.mesos;
+
+import hudson.Extension;
+import hudson.slaves.Cloud;
+import java.util.List;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Mesos {@link Cloud}, which cannot be reconfigured from the web interface.
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+public class NonConfigurableMesosCloud extends MesosCloud {
```

Comment:
What's the UX with the NonConfigurableMesosCloud? Can you show me some screen shots? Is it the case that, once this type of cloud is created, it cannot be re-configured but it can be copied?